config/backend-config/multitenant_workspaces.go

```diff
@@ -134,13 +145,15 @@ func (wc *multiTenantWorkspacesConfig) getFromAPI(ctx context.Context, _ string)
 	writeKeyToWorkspaceIDMap := make(map[string]string)
 	sourceToWorkspaceIDMap := make(map[string]string)
 	workspaceIDToLibrariesMap := make(map[string]LibrariesT)
+	workspaceIDToDataRetentionMap := make(map[string]DataRetention)
 	sourcesJSON := ConfigT{}
 	sourcesJSON.Sources = make([]SourceT, 0)
 	for workspaceID, workspaceConfig := range workspaces.WorkspaceSourcesMap {
 		for _, source := range workspaceConfig.Sources {
 			writeKeyToWorkspaceIDMap[source.WriteKey] = workspaceID
 			sourceToWorkspaceIDMap[source.ID] = workspaceID
 			workspaceIDToLibrariesMap[workspaceID] = workspaceConfig.Libraries
+			workspaceIDToDataRetentionMap[workspaceID] = workspaceConfig.Settings.DataRetention
```

Comment:
@Sidddddarth with respect to changes in #2225, this kind of initialization only happens upon successful retrieval of the config by the http client, whereas we would need these to be initialized if the config was cached too, correct?

That being said, I believe we should initialize them outside this method, as a second step, e.g. maybe expose a `workspaceConfig.ConfigUpdated` method for that purpose to be called by backend-config.

Reply:
```
type ConfigT struct {
	EnableMetrics   bool            `json:"enableMetrics"`
	WorkspaceID     string          `json:"workspaceId"`
	Sources         []SourceT       `json:"sources"`
	Libraries       LibrariesT      `json:"libraries"`
	ConnectionFlags ConnectionFlags `json:"flags"`
}
```

We could change the `Libraries` field to a map(`map[workspaceID]LibrariesT`).
This would remove the need for `GetWorkspaceLibrariesForWorkspaceID(workspaceID)` and other similar functions(ex: `GetWorkspaceIDForSourceID` and `GetWorkspaceIDForWriteKey`) that look redundant really.

That would also address the initialisation problem you describe above.

Reply:
It is not only about libraries, it is about the following exposed interface methods:
https://github.com/rudderlabs/rudder-server/blob/master/config/backend-config/backend-config.go#L51-L53

Changing the struct's fields shouldn't be an option as this would break backwards compatibility of control plane's response schema.

Reply:
> following exposed interface methods

Yes. I'm going thru the calls to these functions, and they're redundant really. We could do without them too.